### ISSUE TITLE
format a Duration

### ISSUE THREAD

**Issue by Eh2406** [chronotope/chrono#197]
I am new to the library But I am not seeing how to format a Duration, nor how to convert to any of the other types with formatting abilities.

**Comment by quodlibetor** [comment]
Our duration is a re-export of [`time::Duration`](https://docs.rs/time/0.1.38/time/struct.Duration.html), which has both `Display` and various `num_*` methods (e.g. [`.num_weeks()`](https://docs.rs/time/0.1.38/time/struct.Duration.html#method.num_weeks) if you want a custom format.

but just using `format!("{}", some_duration)` should do it, formatting as an [ISO Duration](https://en.wikipedia.org/wiki/ISO_8601#Durations). [playground example](https://play.rust-lang.org/)

**Comment by Eh2406** [comment]
The relationship between `time::Duration` vs. `chrono::Duration` was not clear to me, thanks for explaining. The [section that the top of the docs](https://github.com/chronotope/chrono/blob/master/src/lib.rs#L68) explaining it, actually got me more confused. It makes it clear that `time::Duration` is referred to as `OldDuration`, and `newer, standard type for duration,` is referred to as `Duration`. So I assumed that `chrono::Duration` is that `newer, standard type for duration,`

I can definitely roll my own from the `num_*` methods, but from the ISO you linked I may find what I need with text trimming the standard. I think the playground example is just a link to the playground.

I had bean hoping that there would be a `.format("fully custom thing %d:%H:%M:%S")` Like there is for the other types.

**Comment by quodlibetor** [comment]
Yeah actually I think that we are incorrectly exposing `time::Duration` as `chrono::Duration`, it was supposed to be an intermediate step on our way to exposing `std::Duration`. I just forgot :-(

> I had bean hoping that there would be a .format("fully custom thing %d:%H:%M:%S") Like there is for the other types.

We could probably make that happen with our own format trait, and IMO that does belong in this lib. I'll add that as a wishlist item.

**Comment by Eh2406** [comment]
@burkadurka came up with some clever workarounds.
https://www.reddit.com/r/rust/comments/7j09f1/hey_rustaceans_got_an_easy_question_ask_here/drar4ht/

**Comment by quodlibetor** [comment]
That's nifty. I'm tempted to add that as a utility method to chrono as a quick way to use strptime formatting with durations.

**Comment by shaladdle** [comment]
Sorry to necro an old issue, but I wanted to throw in my 2c. @quodlibetor's suggestion of allowing a formatting string would be a step in the right direction, but I think it would be even better to support (in addition to) a more human-readable format, like the one used by Google's `absl::FormatDuration` and `absl::ParseDuration` [here](https://github.com/abseil/abseil-cpp/blob/master/absl/time/time.h#L530-L548). I use this library frequently and it's by far the most easy to use time library I've used in any language

I think a large part of it is because it's easy to print and parse, which are critical for debugging and normal usage of most types. Additionally, command line apps really benefit from being able to parse that format, since it's quite human-readable.

**Comment by Veetaha** [comment]
I wish chrono had means of custom formatting for durations with the `strftime` syntax.
I have a problem porting some code that uses JavaScript's [`moment-duration-format` library](https://www.npmjs.com/package/moment-duration-format#template)

For example something like this JavaScript API with the template string would be very helpful:

```js
moment.duration(123, "minutes").format("h:mm");
// "2:03"
```

**Comment by nkaretnikov** [comment]
I'd suggest at least adding this example to the docs (from the Reddit link posted earlier):
https://play.rust-lang.org/?gist=f8b913eb26a2929ab2adfb4c8a2b6139&version=stable
